apps/lib/opt.c

```diff
@@ -138,6 +138,15 @@ char *opt_progname(const char *argv0)
 }
 #endif
 
+char *opt_appname(const char *arg0)
+{
+    int len = strlen(prog);
+
+    if (arg0 != NULL)
+        snprintf(prog + len, sizeof(prog) - len - 1, " %s", arg0);
+    return prog;
```

Comment:
I think this will break the output in case the openssl binary is symlinked to the command name. In example `ln -s openssl x509`.

I think adding `if (strcmp(prog, arg0) == 0) return prog;` to the beginning would work.

Reply:
Interesting use case - never thought of that.
Yeah, comparing `prog` and `arg0` does the trick to support, e.g.,
```
ln -s apps/openssl x509
PATH=. x509 --help
```